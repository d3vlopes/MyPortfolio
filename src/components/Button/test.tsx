import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Github } from '@styled-icons/boxicons-logos'

import Button from '.'

const { getByRole, getByTestId } = screen

describe('<Button />', () => {
  it('should render by default', () => {
    renderWithTheme(
      <Button variant="primary" size="medium">
        Button
      </Button>,
    )

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#6D59A8',
      'font-size': '2.2rem',
      width: '21.5rem',
      height: '6.2rem',
    })
  })

  it('should render the variant secondary ', () => {
    renderWithTheme(<Button variant="secondary">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#83A848',
    })
  })

  it('should render the variant outline ', () => {
    renderWithTheme(<Button variant="outline">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      border: '1px solid #DEE7EA',
      'background-color': 'transparent',
    })
  })

  it('should render expand version', () => {
    renderWithTheme(<Button expand>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '31.5rem',
      height: '6.2rem',
    })
  })

  it('should render small size', () => {
    renderWithTheme(<Button size="small">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '1.8rem',
      width: '17.5rem',
      height: '5rem',
    })
  })

  it('should render large size', () => {
    renderWithTheme(<Button size="large">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '2.6rem',
      width: '28.8rem',
      height: '7.6rem',
    })
  })

  it('should render an icon version', () => {
    renderWithTheme(
      <Button icon={<Github data-testid="icon" />}>Button</Button>,
    )

    const button = getByRole('button', { name: /button/i })
    const icon = getByTestId('icon')

    expect(button).toBeInTheDocument()
    expect(icon).toBeInTheDocument()
  })

  it('should render Button as a link', () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Button
      </Button>,
    )

    expect(screen.getByRole('link', { name: /button/i })).toHaveAttribute(
      'href',
      '/link',
    )
  })

  it('should up while hover', async () => {
    renderWithTheme(<Button hoverUp>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyleRule('transform', 'translateY(-0.5rem)', {
      modifier: ':hover',
    })
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<Button>Button</Button>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
