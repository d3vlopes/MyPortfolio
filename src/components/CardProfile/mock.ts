import { CardProfileProps } from '.'

export default {
  image: '/img/me.jpg',
  name: 'Leandro Lopes',
  age: 27,
  city: 'Santa Cruz do Sul, RS',
  liking: 'Programação, design, música, séries e filmes.',
  dream:
    'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
} as CardProfileProps
