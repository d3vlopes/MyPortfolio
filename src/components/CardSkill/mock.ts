import { CardSkillProps } from '.'

export default [
  {
    icon: '/img/icons/sword.svg',
    title: 'Desafios',
    description:
      'Estou sempre em busca de novos desafios que me tirem do meu porto seguro.',
  },
  {
    icon: '/img/icons/target-arrow.svg',
    title: 'Persistência',
    description: `Não desanimo em frente aos obstáculos, quanto mais
      difícil, melhor a satisfação.`,
  },
  {
    icon: '/img/icons/lightbulb.svg',
    title: 'Agilidade',
    description: `Tenho facilidade para pensar em possíveis soluções para problemas.`,
  },
  {
    icon: '/img/icons/trending-up.svg',
    title: 'Evolução',
    description: `Gosto muito de aprender coisas novas e estar sempre sempre evoluindo.`,
  },
] as CardSkillProps[]
