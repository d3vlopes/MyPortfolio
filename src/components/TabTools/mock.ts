import { TabToolsProps } from '.'

export default {
  developer: [
    {
      src: '/img/tools/dev/react.svg',
      name: 'React',
    },
    {
      src: '/img/tools/dev/nextjs.svg',
      name: 'NextJS',
    },
    {
      src: '/img/tools/dev/typescript.svg',
      name: 'TypeScript',
    },
    {
      src: '/img/tools/dev/storybook.svg',
      name: 'Storybook',
    },
    {
      src: '/img/tools/dev/jest.svg',
      name: 'Jest',
    },
    {
      src: '/img/tools/dev/testing-library.svg',
      name: 'Testing Library',
    },
    {
      src: '/img/tools/dev/node.svg',
      name: 'Node',
    },
    {
      src: '/img/tools/dev/graphql.svg',
      name: 'GraphQL',
    },
    {
      src: '/img/tools/dev/html.svg',
      name: 'HTML 5',
    },
    {
      src: '/img/tools/dev/css.svg',
      name: 'CSS 3',
    },
    {
      src: '/img/tools/dev/javascript.svg',
      name: 'JavaScript',
    },
    {
      src: '/img/tools/dev/git.svg',
      name: 'Git',
    },
  ],
  design: [
    {
      src: '/img/tools/design/figma.svg',
      name: 'Figma',
    },
    {
      src: '/img/tools/design/photoshop.svg',
      name: 'Photoshop',
    },
    {
      src: '/img/tools/design/illustrator.svg',
      name: 'illustrator',
    },
  ],
} as TabToolsProps
