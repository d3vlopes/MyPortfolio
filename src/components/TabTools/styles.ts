import styled, { css } from 'styled-components'
import { darken } from 'polished'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  width: 100%;
  max-width: 82.3rem;
`

type TabsProps = {
  active: boolean
}

export const Tabs = styled.div<TabsProps>`
  ${({ theme, active }) => css`
    position: relative;
    display: flex;
    justify-content: space-between;
    background-color: ${theme.colors.bg.secondary.dark};
    padding: ${theme.spacings.xsmall} ${theme.spacings.xxlarge};
    transition: all ${theme.transition.fast};

    ${media.lessThan('medium')`
      padding:  ${theme.spacings.xsmall};
    `}

    &::after {
      content: '';
      position: absolute;
      bottom: 0;
      left: 0;
      border-bottom: 0.5rem solid
        ${active ? theme.colors.primary.main : theme.colors.bg.secondary.light};
      width: calc(100% / 2);
    }

    &::before {
      content: '';
      position: absolute;
      bottom: 0;
      right: 0;
      border-bottom: 0.5rem solid
        ${active ? theme.colors.bg.secondary.light : theme.colors.primary.main};
      width: calc(100% / 2);
    }
  `}
`

export const Button = styled.button<TabsProps>`
  ${({ theme, active }) => css`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background: transparent;
    border: none;
    font: normal ${theme.font.weight.medium} ${theme.font.sizes.xxlarge} / 100%
      ${theme.font.body};
    color: ${active ? theme.colors.text.light : theme.colors.grey};
    transition: all ${theme.transition.fast};
    margin-bottom: 0.4rem;
    cursor: pointer;

    ${media.lessThan('medium')`
      font-size: ${theme.font.sizes.small};
    `}

    &:hover {
      color: ${!active && darken(0.2, theme.colors.grey)};
    }

    svg {
      width: 4.5rem;
      height: 4.5rem;
      color: ${active ? theme.colors.white : theme.colors.grey};
      margin-bottom: ${theme.spacings.xsmall};

      ${media.lessThan('medium')`
        width: 3.5rem;
        height: 3.5rem;
    `}
    }
  `}
`

export const Content = styled.div<TabsProps>`
  ${({ theme, active }) => css`
    display: ${active ? 'grid' : 'none'};
    grid-template-columns: repeat(auto-fit, minmax(21rem, 1fr));
    background-color: ${theme.colors.bg.secondary.dark};
    padding: ${theme.spacings.large} ${theme.spacings.xlarge};
    gap: ${theme.spacings.xsmall};
    transition: all ${theme.transition.slow};
  `}
`

export const Tool = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-content: space-between;
    align-items: center;
    gap: ${theme.spacings.xxsmall};

    p {
      font-size: ${theme.font.sizes.small};
      line-height: 150%;
    }
  `}
`
