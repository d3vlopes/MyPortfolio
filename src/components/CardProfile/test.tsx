import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardProfile, { CardProfileProps } from '.'

const props: CardProfileProps = {
  image: '/img/me.jpg',
  name: 'Leandro Lopes',
  age: 27,
  city: 'Santa Cruz do Sul - RS',
  liking: 'Programação, design, música, séries e filmes',
  dream:
    'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
}

const { getByRole, getByText } = screen

describe('<CardProfile />', () => {
  it('should render CardProfile', () => {
    const { container } = renderWithTheme(<CardProfile {...props} />)

    const image = getByRole('img', { name: 'Leandro Lopes' })
    const name = getByText('Leandro Lopes')
    const age = getByText('27 anos')
    const city = getByText('Santa Cruz do Sul - RS')
    const liking = getByText('Programação, design, música, séries e filmes')
    const dream = getByText(
      'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
    )

    expect(image).toHaveAttribute('src', '/img/me.jpg')
    expect(name).toBeInTheDocument()
    expect(age).toBeInTheDocument()
    expect(city).toBeInTheDocument()
    expect(liking).toBeInTheDocument()
    expect(dream).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
