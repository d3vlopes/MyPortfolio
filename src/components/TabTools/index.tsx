import Image, { ImageLoaderProps } from 'next/image'
import { useState, Fragment } from 'react'

import { CodeCurly } from '@styled-icons/boxicons-regular/CodeCurly'
import { Palette } from '@styled-icons/boxicons-solid/Palette'

import * as S from './styles'

type Tool = {
  src: string
  name: string
}

export type TabToolsProps = {
  developer: Tool[]
  design: Tool[]
}

const TabTools = ({ developer, design }: TabToolsProps) => {
  const [isOpen, setIsOpen] = useState(true)

  const handleOpen = () => {
    setIsOpen(!isOpen)
  }

  const Loader = ({ src, width, quality }: ImageLoaderProps) => {
    return `${src}?w=${width}&q=${quality || 75}`
  }

  const renderTools = (tool: Tool) => {
    return (
      <Fragment key={`tool-${tool.name}`}>
        <S.Tool>
          <Image
            loader={Loader}
            src={tool.src}
            alt={tool.name}
            width={35}
            height={35}
          />
          <p>{tool.name}</p>
        </S.Tool>
      </Fragment>
    )
  }

  return (
    <S.Wrapper>
      <S.Tabs active={isOpen}>
        <S.Button active={isOpen} onClick={handleOpen} disabled={isOpen}>
          <CodeCurly data-testid="icon-development" /> Development
        </S.Button>

        <S.Button active={!isOpen} onClick={handleOpen} disabled={!isOpen}>
          <Palette data-testid="icon-design" /> Design
        </S.Button>
      </S.Tabs>

      <S.Content
        active={isOpen}
        aria-hidden={!isOpen}
        data-testid="content-development"
      >
        {developer.map(renderTools)}
      </S.Content>
      <S.Content
        active={!isOpen}
        aria-hidden={isOpen}
        data-testid="content-design"
      >
        {design.map(renderTools)}
      </S.Content>
    </S.Wrapper>
  )
}

export default TabTools
