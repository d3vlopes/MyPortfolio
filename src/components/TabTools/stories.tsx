import { Story, Meta } from '@storybook/react'
import TabTools, { TabToolsProps } from '.'
import mock from './mock'

export default {
  title: 'Components/TabTools',
  component: TabTools,
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/KIQWZ47GDrrNCIxysbEqOA/My-Portfolio?node-id=657%3A1048',
    },
  },
  args: {
    developer: mock.developer,
    design: mock.design,
  },
} as Meta<TabToolsProps>

export const Default: Story<TabToolsProps> = (args) => {
  return (
    <div
      style={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
      }}
    >
      <TabTools {...args} />
    </div>
  )
}
