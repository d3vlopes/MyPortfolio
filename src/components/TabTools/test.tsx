import { screen, fireEvent } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import TabTools from '.'
import mock from './mock'

const { getByTestId, getByRole } = screen

const tools = {
  developer: mock.developer,
  design: mock.design,
}

describe('<TabTools />', () => {
  it('should render TabTools', () => {
    renderWithTheme(<TabTools {...tools} />)

    const iconDevelopment = getByTestId('icon-development')
    const iconDesign = getByTestId('icon-design')

    const developmentButton = getByRole('button', { name: 'Development' })
    const designButton = getByRole('button', { name: 'Design' })

    const developmentContent = getByTestId('content-development')
    const designContent = getByTestId('content-design')

    expect(iconDevelopment).toBeInTheDocument()
    expect(iconDesign).toBeInTheDocument()
    expect(developmentButton).toBeInTheDocument()
    expect(designButton).toBeInTheDocument()

    expect(developmentContent.getAttribute('aria-hidden')).toBe('false')
    expect(designContent.getAttribute('aria-hidden')).toBe('true')

    expect(developmentContent).toHaveStyle({
      display: 'flex',
    })
    expect(designContent).toHaveStyle({
      display: 'none',
    })
  })

  it('should change tab active on click', () => {
    renderWithTheme(<TabTools {...tools} />)

    const developmentButton = getByRole('button', { name: 'Development' })
    const designButton = getByRole('button', { name: 'Design' })

    const developmentContent = getByTestId('content-development')
    const designContent = getByTestId('content-design')

    fireEvent.click(designButton)

    expect(developmentContent.getAttribute('aria-hidden')).toBe('true')
    expect(designContent.getAttribute('aria-hidden')).toBe('false')

    expect(developmentContent).toHaveStyle({
      display: 'none',
    })
    expect(designContent).toHaveStyle({
      display: 'flex',
    })

    fireEvent.click(developmentButton)

    expect(developmentContent.getAttribute('aria-hidden')).toBe('false')
    expect(designContent.getAttribute('aria-hidden')).toBe('true')
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<TabTools {...tools} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
