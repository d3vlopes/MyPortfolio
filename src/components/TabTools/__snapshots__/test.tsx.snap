// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabTools /> should match snapshot 1`] = `
.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  width: 100%;
  max-width: 82.3rem;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #302F43;
  padding: 1.6rem 8rem;
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  border-bottom: 0.5rem solid #6D59A8;
  width: calc(100% / 2);
}

.c1::before {
  content: '';
  position: absolute;
  bottom: 0;
  right: 0;
  border-bottom: 0.5rem solid #484762;
  width: calc(100% / 2);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: none;
  font: normal 500 2.6rem / 100% Rubik,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  color: #FFFFFF;
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
  margin-bottom: 0.4rem;
  cursor: pointer;
}

.c2 svg {
  width: 4.5rem;
  height: 4.5rem;
  color: #FFFFFF;
  margin-bottom: 1.6rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: none;
  font: normal 500 2.6rem / 100% Rubik,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  color: rgba(255,255,255,0.5);
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
  margin-bottom: 0.4rem;
  cursor: pointer;
}

.c4:hover {
  color: rgba(204,204,204,0.5);
}

.c4 svg {
  width: 4.5rem;
  height: 4.5rem;
  color: rgba(255,255,255,0.5);
  margin-bottom: 1.6rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  background-color: #302F43;
  padding: 4.0rem 0;
  gap: 2.4rem;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
}

.c8 {
  display: none;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  background-color: #302F43;
  padding: 4.0rem 0;
  gap: 2.4rem;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.6rem;
  margin-top: 1.6rem;
}

.c6 p {
  font-size: 1.8rem;
  line-height: 150%;
}

.c7 {
  width: 3.5rem;
  height: 3.5rem;
  max-width: 100%;
}

@media (max-width:768px) {
  .c1 {
    padding: 1.6rem;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

@media (max-width:768px) {
  .c2 {
    font-size: 1.8rem;
  }
}

@media (max-width:768px) {
  .c2 svg {
    width: 3.5rem;
    height: 3.5rem;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

@media (max-width:768px) {
  .c4 svg {
    width: 3.5rem;
    height: 3.5rem;
  }
}

@media (max-width:450px) {
  .c5 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    padding: 4.0rem 1.6rem;
  }

  .c5 .content-design {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    padding: 4.0rem 0;
  }
}

@media (max-width:450px) {
  .c8 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    padding: 4.0rem 1.6rem;
  }

  .c8 .content-design {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    padding: 4.0rem 0;
  }
}

@media (max-width:768px) {
  .c6 p {
    font-size: 1.6rem;
  }
}

@media (max-width:450px) {
  .c6 p {
    font-size: 1.4rem;
  }
}

@media (max-width:768px) {
  .c7 {
    width: 3rem;
    height: 3rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <button
      class="c2"
      disabled=""
    >
      <svg
        aria-hidden="true"
        class="c3"
        data-testid="icon-development"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 22h1v-2h-.989C8.703 19.994 6 19.827 6 16c0-1.993-.665-3.246-1.502-4C5.335 11.246 6 9.993 6 8c0-3.827 2.703-3.994 3-4h1V2H8.998C7.269 2.004 4 3.264 4 8c0 2.8-1.678 2.99-2.014 3L2 13c.082 0 2 .034 2 3 0 4.736 3.269 5.996 5 6zm13-11c-.082 0-2-.034-2-3 0-4.736-3.269-5.996-5-6h-1v2h.989c.308.006 3.011.173 3.011 4 0 1.993.665 3.246 1.502 4-.837.754-1.502 2.007-1.502 4 0 3.827-2.703 3.994-3 4h-1v2h1.002C16.731 21.996 20 20.736 20 16c0-2.8 1.678-2.99 2.014-3L22 11z"
        />
      </svg>
       Development
    </button>
    <button
      class="c4"
    >
      <svg
        aria-hidden="true"
        class="c3"
        data-testid="icon-design"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.38 21.646A9.985 9.985 0 0 0 12 22l.141-.001a2.998 2.998 0 0 0 2.515-1.425c.542-.876.6-1.953.153-2.88l-.198-.415c-.453-.942-.097-1.796.388-2.281.485-.485 1.341-.841 2.28-.388h.001l.413.199a2.99 2.99 0 0 0 2.881-.153A2.997 2.997 0 0 0 22 12.141a9.926 9.926 0 0 0-.353-2.76c-1.038-3.827-4.353-6.754-8.246-7.285-3.149-.427-6.241.602-8.471 2.833S1.666 10.247 2.096 13.4c.53 3.894 3.458 7.208 7.284 8.246zM15.5 6a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3zm-5-1a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3zM9 15.506a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm-2.5-6.5a1.5 1.5 0 1 1-.001 3.001A1.5 1.5 0 0 1 6.5 9.006z"
        />
      </svg>
       Design
    </button>
  </div>
  <div
    aria-hidden="false"
    class="c5"
    data-testid="content-development"
  >
    <div>
      <div
        class="c6"
      >
        <img
          alt="React"
          class="c7"
          src="/img/tools/dev/react.svg"
        />
        <p>
          React
        </p>
      </div>
      <div
        class="c6"
      >
        <img
          alt="NextJS"
          class="c7"
          src="/img/tools/dev/nextjs.svg"
        />
        <p>
          NextJS
        </p>
      </div>
      <div
        class="c6"
      >
        <img
          alt="TypeScript"
          class="c7"
          src="/img/tools/dev/typescript.svg"
        />
        <p>
          TypeScript
        </p>
      </div>
      <div
        class="c6"
      >
        <img
          alt="Storybook"
          class="c7"
          src="/img/tools/dev/storybook.svg"
        />
        <p>
          Storybook
        </p>
      </div>
      <div
        class="c6"
      >
        <img
          alt="Jest"
          class="c7"
          src="/img/tools/dev/jest.svg"
        />
        <p>
          Jest
        </p>
      </div>
      <div
        class="c6"
      >
        <img
          alt="Testing Library"
          class="c7"
          src="/img/tools/dev/testing-library.svg"
        />
        <p>
          Testing Library
        </p>
      </div>
    </div>
    <div>
      <div
        class="c6"
      >
        <img
          alt="Node"
          class="c7"
          src="/img/tools/dev/node.svg"
        />
        <p>
          Node
        </p>
      </div>
      <div
        class="c6"
      >
        <img
          alt="GraphQL"
          class="c7"
          src="/img/tools/dev/graphql.svg"
        />
        <p>
          GraphQL
        </p>
      </div>
      <div
        class="c6"
      >
        <img
          alt="HTML 5"
          class="c7"
          src="/img/tools/dev/html.svg"
        />
        <p>
          HTML 5
        </p>
      </div>
      <div
        class="c6"
      >
        <img
          alt="CSS 3"
          class="c7"
          src="/img/tools/dev/css.svg"
        />
        <p>
          CSS 3
        </p>
      </div>
      <div
        class="c6"
      >
        <img
          alt="JavaScript"
          class="c7"
          src="/img/tools/dev/javascript.svg"
        />
        <p>
          JavaScript
        </p>
      </div>
      <div
        class="c6"
      >
        <img
          alt="Git"
          class="c7"
          src="/img/tools/dev/git.svg"
        />
        <p>
          Git
        </p>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="c8 content-design"
    data-testid="content-design"
  >
    <div
      class="c6"
    >
      <img
        alt="Figma"
        class="c7"
        src="/img/tools/design/figma.svg"
      />
      <p>
        Figma
      </p>
    </div>
    <div
      class="c6"
    >
      <img
        alt="Photoshop"
        class="c7"
        src="/img/tools/design/photoshop.svg"
      />
      <p>
        Photoshop
      </p>
    </div>
    <div
      class="c6"
    >
      <img
        alt="illustrator"
        class="c7"
        src="/img/tools/design/illustrator.svg"
      />
      <p>
        illustrator
      </p>
    </div>
  </div>
</div>
`
